hi-ml/src/health_ml/utils/checkpoint_handler.py

```diff
@@ -69,10 +76,56 @@ def get_checkpoint_to_test(self) -> Path:
             if checkpoint_from_current_run.is_file():
                 logging.info(f"Using checkpoint from current run: {checkpoint_from_current_run}")
                 return checkpoint_from_current_run
-            else:
-                raise FileNotFoundError(f"Checkpoint file does not exist: {checkpoint_from_current_run}")
+            logging.warning(f"No inference checkpoint found from the current run: {checkpoint_from_current_run}")
+            # Handling low-priority preemption: If we are recovering from a preemption that appeared after training was
```

Comment:
Personal taste on styling here: I think multi-line comments like this should be avoided unless absolutely necessary. It makes the code much more difficult to read. The logic explained here should either be self-evident from the code (+ simple, single-line comments) or explained in the function docstrings, not using multi-line comments in the code itself.

Reply:
I disagree on avoiding multiline comments. Code cannot always speak for itself - this is amplified as time passes, and if you're new to a codebase. What seems self-evident for the author is usually not for someone else.
Good point here though on moving the described scenario to the docstring, it is important to have it there..